sdk/src/main/scala/com/horizen/SidechainApp.scala

```diff
@@ -372,4 +385,56 @@ class SidechainApp @Inject()
   def getSecretSubmitProvider: SecretSubmitProvider = secretSubmitProvider
 
   actorSystem.eventStream.publish(SidechainAppEvents.SidechainApplicationStart)
+
+  val timeoutDuration: FiniteDuration = settings.restApi.timeout
+
+  implicit val timeout: Timeout = Timeout(timeoutDuration)
+
+  /***
+   * Retrieve the SidechainBlockId needed to rollback the SidechainStateStorage for the backup.
+   * It's calculated by the following formula:
+   * Genesis_MC_block_height + (current_epch-2) * withdrawalEpochçength -1
+   * @param view
+   * @return sidechainBlockId to rollback or Exception if it's not found
+   */
+  def getSidechainBlockIdForBackUpRollback(view: View):Array[Byte] = {
+    val currentEpoch = view.state.getWithdrawalEpochInfo.epoch
+    val genesisMcBlockHeight = view.history.getMainchainCreationBlockHeight
+    val withdrawalEpochLength = sidechainSettings.genesisData.withdrawalEpochLength
+    val blockHeightToRollback = genesisMcBlockHeight + (currentEpoch -2) * withdrawalEpochLength - 1
+    val mainchainBlockReferenceInfo = view.history.getMainchainBlockReferenceInfoByMainchainBlockHeight(blockHeightToRollback).get()
+    mainchainBlockReferenceInfo.getMainchainReferenceDataSidechainBlockId
+  }
+
+  def createBackup(): Unit = {
+    log.info(s"Starting sidechain backup...")
+
+    val checkAsFuture = (nodeViewHolderRef ? GetDataFromCurrentView(getSidechainBlockIdForBackUpRollback)).asInstanceOf[Future[Array[Byte]]]
+    checkAsFuture.onComplete{
+      case Success(blockId) =>
+        log.info(s"Rollback of the SidechainStateStorage to version: ${BytesUtils.toHexString(blockId)}")
+        sidechainStateStorage.rollback(new ByteArrayWrapper(blockId)) match {
```

Comment:
Can we avoid to broke the node state by persistent rollback operation?
What if the process will be interrupted in the middle?
I would suggest to add the flag like "copyBeforeRollback" with default value `true`, that will copy DB folder first. We need the flag, because the user may use the original storage as something that we can't copy (remote cloud instance, for example), but it is very unlikely.

Reply:
Done